redhat/lyrionmusicserver.spec

```
@@ -356,7 +308,7 @@ function parseSysconfigSqueezeboxserver {
 
 	# Check if any additions to the LYRION_ARGS variable have been made.
 	# Do that by filter out the ones we know should be there.
-	extra=`echo $LYRION_ARGS |tr " " "\n"|grep -v -E "(--daemon|--prefsdir|--logdir|--cachedir|--charset)"` || :
+	extra=`echo $LYRION_ARGS |/usr/bin/tr " " "\n"|/usr/bin/perl -lane "/(--daemon|--prefsdir|--logdir|--cachedir|--charset)/i or print"` || :
```

We can eliminate the `tr` here, too:

```suggestion
	extra=`echo $LYRION_ARGS | perl -lane 'print foreach grep { not m/^(?:--daemon|--prefsdir|--logdir|--cachedir|--charset)(?:=|$)/  } @F' || :
```

---

Yes we can get rid of tr, I think the readability of the code suffers though. And as tr is in the same RPM package as cp, mv, rm (coreutils) it wouldn't reduce the dependency on other packages.

It might of course be a tiny bit faster to execute. is that the reason you want to remove it.